features/networking/service.feature

```diff
@@ -153,3 +153,433 @@ Feature: Service related networking scenarios
     Then the step should succeed
     And the output should contain:
       | 8080 |
+
+  # @author weliang@redhat.com
+  # @case_id OCP-24668
+  @admin
+  Scenario: externalIP defined in service but no sepc.externalIP defined	
+    Given I switch to the first user
+    Given I have a project 
+    And evaluation of `project.name` is stored in the :proj1 clipboard
+    
+    # Create a service with a externalIP in :proj1
+    When I run the :create client command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/externalip_service1.json | 
+      | n | <%= cb.proj1 %> |
+    Then the output should contain "externalIPs have been disabled" 
+    
+    
+  # @author weliang@redhat.com
+  # @case_id OCP-24669
+  @admin
+  Scenario: externalIP defined in service with set ExternalIP in allowedCIDRs	 
+    # Create additional network through CNO
+    When I run the :patch admin command with:
+      | resource      | networks.config.openshift.io |
+      | resource_name | cluster           |
+      | p             | {"spec":{"externalIP":{"policy":{"allowedCIDRs":["22.2.2.0/24"]}}}} |
+      | type          | merge | 
+    Then I wait for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource      | networks.config.openshift.io                |
+      | resource_name | cluster             |
+      | template      | {{{{.spec.externalIP.policy.allowedCIDRs}}}} |
+      | o             | yaml   |
+    And the output should contain "22.2.2.0"
+    """
+ 
+    # Need wait for about 300 seconds to apply new additional network configuration
+    Given 300 seconds have passed
```

Comment:
still need wait 300 seconds when using above step line 182-191

Reply:
Yes, we need wait 300 seconds for each test case, because we update CNO configuration in each test case

Reply:
Usually we wait for some event to happen instead of hardcoding wait. In rare situations waiting is necessary. So is there an event we can check for instead of hardcoding wait? If there is no such event, what are these 300 seconds based on? Configuration polling? number of nodes? Anything?

Reply:
After updating configuration in CNO, right now we do not have a correct way to find out how long it will take to apply new configuration in the cluster, I confirmed this with developers. The testing start to fail randomly when using 300 seconds in latest image, so I am using 400 seconds now. In the future this script can be updated if we find a way to check whether cluster is reay with new CNO configuration and not using 400 seconds